morpheus/_lib/src/python_modules/messages.cpp

```diff
@@ -73,6 +77,14 @@ PYBIND11_MODULE(messages, m)
     // Allows python objects to keep DataTable objects alive
     py::class_<IDataTable, std::shared_ptr<IDataTable>>(m, "DataTable");
 
+    srf::pysrf::PortBuilderUtil::register_port_util<std::shared_ptr<MessageMeta>>();
+    srf::pysrf::PortBuilderUtil::register_port_util<std::shared_ptr<MultiMessage>>();
+    srf::pysrf::PortBuilderUtil::register_port_util<std::shared_ptr<MultiInferenceMessage>>();
+    srf::pysrf::PortBuilderUtil::register_port_util<std::shared_ptr<MultiInferenceFILMessage>>();
+    srf::pysrf::PortBuilderUtil::register_port_util<std::shared_ptr<MultiInferenceNLPMessage>>();
+    srf::pysrf::PortBuilderUtil::register_port_util<std::shared_ptr<MultiResponseMessage>>();
+    srf::pysrf::PortBuilderUtil::register_port_util<std::shared_ptr<MultiResponseProbsMessage>>();
```

Comment:
That registers Ingress/Egress port builders for that specific data type that can be looked up via typeid. What this `should` allow us to do is try and match the `cpptype` typeid of a datatype specified in the port_id_tuple with a registered port constructor. The effect is that instead of creating a new Egress/Ingress port of type a py::object we create one of the actual c++ type and avoid having to go into python land when moving/accessing the object.

Reply:
This is slightly broken, in morpheus, at the moment because reliably getting the cpptype from a pybind11 wrapped object defined in Morpheus isn't exactly the same as doing it in SRF (because of different compilation units). So as of now, the lookup will fail and fall back to using a py::object instead of the c++ class.